### ISSUE TITLE
Fails running custom initialization SQL

### ISSUE THREAD

**Issue by steven-sheehy** [bitnami/bitnami-docker-postgresql-repmgr#23]
**Description**

A recent change broke running custom initialization SQL. It no longer will propogate the password for the postgres user to the code that runs the init SQL. I believe it's due to this [commit](https://github.com/bitnami/bitnami-docker-postgresql-repmgr/commit/429b997b5799cce8cf610513ff3899d8793f2034#diff-7788a82d515c18c5ca9bc103f8ef41b3R218) not passing `POSTGRESQL_PASSWORD` as the [old code](https://github.com/bitnami/bitnami-docker-postgresql-repmgr/commit/429b997b5799cce8cf610513ff3899d8793f2034#diff-581841e951673be4f9d0d8b0879277afL201) did.

12.3.0-debian-10-r36 does not work
12.3.0-debian-10-r35 does work

**Steps to reproduce the issue:**

1. Run postgresql-ha with a custom sql in a secret

**Describe the results you received:**

PostgreSQL will startup but skip the initialization SQL.

**Describe the results you expected:**

PostgreSQL can run custom initialization scripts on first start.

**Additional information you deem important (e.g. issue happens only occasionally):**

```
postgresql-repmgr 20:59:30.52
postgresql-repmgr 20:59:30.53 Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 20:59:30.53 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
postgresql-repmgr 20:59:30.53 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
postgresql-repmgr 20:59:30.53
postgresql-repmgr 20:59:30.55 DEBUG ==> Configuring libnss_wrapper...
postgresql-repmgr 20:59:30.58 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 20:59:30.60 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 20:59:30.60 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 20:59:30.61 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 20:59:30.62 DEBUG ==> Checking node 'marketplace-db-postgresql-0.marketplace-db-postgresql-headless.marketplace.svc.cluster.local:5432'...
psql: could not connect to server: Connection refused
	Is the server running on host "marketplace-db-postgresql-0.marketplace-db-postgresql-headless.marketplace.svc.cluster.local" (10.12.5.246) and accepting
	TCP/IP connections on port 5432?
postgresql-repmgr 20:59:30.64 DEBUG ==> Skipping: failed to get primary from the node 'marketplace-db-postgresql-0.marketplace-db-postgresql-headless.marketplace.svc.cluster.local:5432'!
postgresql-repmgr 20:59:30.64 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 20:59:30.66 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 20:59:30.70 DEBUG ==> Injecting a new postgresql.conf file...
postgresql-repmgr 20:59:30.70 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql-repmgr 20:59:30.79 DEBUG ==> Injecting a new pg_hba.conf file...
postgresql-repmgr 20:59:30.80 INFO  ==> Preparing repmgr configuration...
postgresql-repmgr 20:59:30.81 DEBUG ==> Node ID: '1000', Rol: 'primary', Primary Node: ':5432'
postgresql-repmgr 20:59:30.81 INFO  ==> Initializing Repmgr...
postgresql-repmgr 20:59:30.81 INFO  ==> Initializing PostgreSQL database...
postgresql-repmgr 20:59:30.82 DEBUG ==> Copying files from /bitnami/postgresql/conf to /opt/bitnami/postgresql/conf
postgresql-repmgr 20:59:30.83 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql-repmgr 20:59:30.83 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
postgresql-repmgr 20:59:30.83 DEBUG ==> Ensuring expected directories/files exist...
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.UTF-8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /bitnami/postgresql/data ... ok
creating subdirectories ... ok
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default timezone ... Etc/UTC
selecting dynamic shared memory implementation ... posix
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

Success. You can now start the database server using:

    /opt/bitnami/postgresql/bin/pg_ctl -D /bitnami/postgresql/data -l logfile start


WARNING: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
postgresql-repmgr 20:59:32.64 INFO  ==> Starting PostgreSQL in background...
waiting for server to start....2020-06-23 20:59:32.703 GMT [104] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2020-06-23 20:59:32.703 GMT [104] LOG:  could not bind IPv6 address "::1": Cannot assign requested address
2020-06-23 20:59:32.703 GMT [104] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2020-06-23 20:59:32.709 GMT [104] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2020-06-23 20:59:32.722 GMT [104] LOG:  redirecting log output to logging collector process
2020-06-23 20:59:32.722 GMT [104] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
2020-06-23 20:59:32.727 GMT [106] LOG:  database system was shut down at 2020-06-23 20:59:32 GMT
2020-06-23 20:59:32.736 GMT [104] LOG:  database system is ready to accept connections
 done
server started
postgresql-repmgr 20:59:32.80 INFO  ==> Changing password of postgres
ALTER ROLE
postgresql-repmgr 20:59:32.84 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 20:59:32.86 INFO  ==> Creating repmgr user: repmgr
CREATE ROLE
ALTER ROLE
ALTER ROLE
postgresql-repmgr 20:59:32.92 INFO  ==> Creating repmgr database: repmgr
CREATE DATABASE
postgresql-repmgr 20:59:33.20 INFO  ==> Stopping PostgreSQL...
2020-06-23 20:59:33.213 GMT [104] LOG:  received smart shutdown request
2020-06-23 20:59:33.218 GMT [104] LOG:  background worker "logical replication launcher" (PID 113) exited with exit code 1
2020-06-23 20:59:33.219 GMT [107] LOG:  shutting down
2020-06-23 20:59:33.370 GMT [104] LOG:  database system is shut down
postgresql-repmgr 20:59:34.22 INFO  ==> Starting PostgreSQL in background...
waiting for server to start....2020-06-23 20:59:34.271 GMT [162] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2020-06-23 20:59:34.271 GMT [162] LOG:  listening on IPv6 address "::", port 5432
2020-06-23 20:59:34.276 GMT [162] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2020-06-23 20:59:34.290 GMT [162] LOG:  redirecting log output to logging collector process
2020-06-23 20:59:34.290 GMT [162] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
2020-06-23 20:59:34.296 GMT [164] LOG:  database system was shut down at 2020-06-23 20:59:33 GMT
2020-06-23 20:59:34.305 GMT [162] LOG:  database system is ready to accept connections
 done
server started
postgresql-repmgr 20:59:34.37 INFO  ==> Registering Primary...
INFO: connecting to primary database...
DEBUG: connecting to: "user=repmgr password=password connect_timeout=5 dbname=repmgr host=marketplace-db-postgresql-0.marketplace-db-postgresql-headless.marketplace.svc.cluster.local port=5432 fallback_application_name=repmgr"
NOTICE: attempting to install extension "repmgr"
NOTICE: "repmgr" extension successfully installed
INFO: executing notification command for event "cluster_created"
DETAIL: command is:
  /opt/bitnami/repmgr/events/router.sh 1000 cluster_created 1 "2020-06-23 20:59:34.471281+00" ""
[REPMGR EVENT] Node id: 1000; Event type: cluster_created; Success [1|0]: 1; Time: 2020-06-23 20:59:34.471281+00;  Details:
Looking for the script: /opt/bitnami/repmgr/events/execs/cluster_created.sh
[REPMGR EVENT] no script '/opt/bitnami/repmgr/events/execs/cluster_created.sh' found. Skipping...
INFO: executing notification command for event "primary_register"
DETAIL: command is:
  /opt/bitnami/repmgr/events/router.sh 1000 primary_register 1 "2020-06-23 20:59:34.519763+00" ""
[REPMGR EVENT] Node id: 1000; Event type: primary_register; Success [1|0]: 1; Time: 2020-06-23 20:59:34.519763+00;  Details:
Looking for the script: /opt/bitnami/repmgr/events/execs/primary_register.sh
[REPMGR EVENT] will execute script '/opt/bitnami/repmgr/events/execs/primary_register.sh' for the event
[REPMGR EVENT::primary_register] Node id: 1000; Event type: primary_register; Success [1|0]: 1; Time: 2020-06-23 20:59:34.519763+00;  Details:
[REPMGR EVENT::primary_register] Locking primary...
[REPMGR EVENT::primary_register] Unlocking standby...
NOTICE: primary node record (ID: 1000) registered
postgresql-repmgr 20:59:34.57 INFO  ==> Loading custom scripts...
postgresql-repmgr 20:59:34.58 INFO  ==> Loading user's custom files from /docker-entrypoint-initdb.d ...
postgresql-repmgr 20:59:34.58 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 20:59:34.60 DEBUG ==> Executing /docker-entrypoint-initdb.d/secret/..2020_06_23_20_59_00.298184574/init.sql
Password for user postgres:
2020-06-23 20:59:34.617 GMT [197] FATAL:  password authentication failed for user "postgres"
2020-06-23 20:59:34.617 GMT [197] DETAIL:  Password does not match for user "postgres".
	Connection matched pg_hba.conf line 4: "host     all              all       0.0.0.0/0    md5"
psql: FATAL:  password authentication failed for user "postgres"
postgresql-repmgr 20:59:34.62 INFO  ==> Stopping PostgreSQL...
2020-06-23 20:59:34.633 GMT [162] LOG:  received smart shutdown request
2020-06-23 20:59:34.640 GMT [162] LOG:  background worker "logical replication launcher" (PID 171) exited with exit code 1
2020-06-23 20:59:34.641 GMT [165] LOG:  shutting down
2020-06-23 20:59:34.761 GMT [162] LOG:  database system is shut down
```

**Version**

- Output of `docker version`:

```
(paste your output here)
```

- Output of `docker info`:

```
(paste your output here)
```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
GKE

**Comment by marcosbc** [comment]
Hi @steven-sheehy, you're right. We've identified the error and are working on publishing a fix. We'll keep you posted.

**Comment by steven-sheehy** [comment]
@marcosbc Thanks for checking. One other related thing I saw is that when the initialization script failed due to this bug, it would restart and on the next startup the initialization scripts would not be attempted. It should continue to retry the initialization scripts each start until they are successful.

**Comment by marcosbc** [comment]
Just wanted to let you know that this was fixed in `12.3.0-debian-10-r43`. Thanks for reporting this!

>One other related thing I saw is that when the initialization script failed due to this bug, it would restart and on the next startup the initialization scripts would not be attempted. It should continue to retry the initialization scripts each start until they are successful.

The current logic will attempt to execute the scripts until an execution where all init scripts have succeeded. You can see it here: https://github.com/bitnami/bitnami-docker-postgresql-repmgr/blob/master/12/debian-10/rootfs/opt/bitnami/scripts/libpostgresql.sh#L577 That means that as soon as all init scripts have been executed once, in a next run no init scripts will be run (despite having added new ones since the last execution).
